src/aws-cpp-sdk-core/source/client/AWSClient.cpp

```diff
@@ -791,56 +792,69 @@ void AWSClient::AddChecksumToRequest(const std::shared_ptr<Aws::Http::HttpReques
                               request.GetServiceSpecificParameters()->parameterMap.find("overrideChecksumDisable") !=
                               request.GetServiceSpecificParameters()->parameterMap.end();
 
-    // Request checksums
+    //Check if user has provided the checksum algorithm
     if (!checksumAlgorithmName.empty() && !shouldSkipChecksum)
     {
+        // Check if user has provided a checksum value for the specified algorithm
+        const Aws::String checksumType = "x-amz-checksum-" + checksumAlgorithmName;
+        const HeaderValueCollection &headers = request.GetHeaders();
+        bool checksumValueAndAlgorithmProvided = headers.find(checksumType) != headers.end();
+
         // For non-streaming payload, the resolved checksum location is always header.
         // For streaming payload, the resolved checksum location depends on whether it is an unsigned payload, we let AwsAuthSigner decide it.
-        if (checksumAlgorithmName == "crc32")
+        if (request.IsStreaming() && checksumValueAndAlgorithmProvided)
+        {
+            auto hash = Aws::MakeShared<Crypto::PrecalculatedHash>(AWS_CLIENT_LOG_TAG,request.GetHeaders().find(checksumType)->second);
```

Comment:
nit: spacing inconsistent, do we need to run lint?

Reply:
fixed spacing, we really should have a clang format file. We've discussed it multiple times and i think the only thing stopping us is the one time format the world event. Im very much pro a format the world event though. but we should start thinking about adding one sooner than later.